gcloud/storage/_implicit_environ.py

```diff
@@ -81,29 +29,11 @@ class _DefaultsContainer(object):
     :param connection: Persistent implied connection from environment.
     """
 
-    @_lazy_property_deco
-    @staticmethod
-    def project():
-        """Return the implicit default project."""
-        return _determine_default_project()
-
-    def __init__(self, project=None, bucket=None, connection=None,
-                 implicit=False):
-        if project is not None or not implicit:
-            self.project = project
+    def __init__(self, bucket=None, connection=None):
```

Comment:
ISTM that the `_DefaultsContainer` is an implementation detail of the API functions which use it:  once the `get_default_project()` / `set_default_project()` functions move up to `gcloud._helpers` (so that they can be re-used by `pubsub`), they need their "own" container.  `storage.set_defaults` takes a `project` paramteer, but delegates setting "up" to the APIs in `_helpers`:  that seems like enough of a signal to me.